src/routes/docs/tutorials/pagination/step-7/+page.markdoc

```diff
@@ -0,0 +1,184 @@
+        TODOS_COLLECTION_ID,
+        [
+          Query.orderDesc("$createdAt"),
+          Query.limit(pageLimit),
+          Query.offset(offset),
+        ],
+      );
+
+      setTasks(response.documents);
+      setCurrentPage(Math.floor(offset / pageLimit) + 1);
+      setCounts({
+        totalPages: response.total / pageLimit,
+        totalResults: response.total,
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  }
+  useEffect(() => {
+    init();
+  }, [offset]);
+
+  return (
+    <TasksContext.Provider
+      value={{
+        current: tasks,
+        handleNextPage,
+        handlePreviousPage,
+        currentPage,
+        offset,
+      }}>
+      {props.children}
+    </TasksContext.Provider>
+  );
+}
+```
+
+## Tasks page {% #tasks-page %}
+
+For offset pagination in our app, we need "forward" and "back" buttons to navigate our dataset. We also want to display the current page. The Tasks page will contain the following UI, with navigation handled by the `useTasks` hook:
+* a "back" button that will handle subtracting a given page limit from the offset and disabled when at the beginning of the dataset.
+* a span tag that will show the current page.
+* a "forward" button that will handle adding a given page limit to the offset and disabled when we reach the end of the dataset.
+
+Update the Tasks page in `src/pages/Tasks.jsx` to the following:
+
+```js
+import { useEffect, useState, useCallback } from "react";
+import { useTasks } from "../lib/context/tasks";
+export function Tasks() {
+  const tasks = useTasks();
+  const pageLimit = 4;
+
+  return (
+    <section>
+      <h2>Todos</h2>
+      <ul>
+        {tasks.current &&
+          tasks.current.map((task) => (
+            <li key={`task-${task.taskId}`}>
+              <span>{`Task-${task.taskId}: `}</span>
+              <input type='checkbox' id={task.taskId} name={task.title} />
+              <label htmlFor={task.title}>{task.title}</label>
+            </li>
+          ))}
+      </ul>
+      <Pagination
+        totalTasks={tasks.totalTasks}
+        tasksPerPage={pageLimit}
+        nextOffset={tasks.offset + pageLimit}
+        handleNextPage={tasks.handleNextPage}
+        handlePreviousPage={tasks.handlePreviousPage}
+      />
+    </section>
+  );
+}
+
+function Pagination(props) {
+  const tasks = useTasks();
+  const pageLimit = 4;
+
+  const {
+    totalTasks,
+    tasksPerPage,
+    nextOffset,
+    handleNextPage,
+    handlePreviousPage,
+  } = props;
+  return (
+    <div>
+      <button
+        disabled={tasks.currentPage === 1}
+        onClick={() => handlePreviousPage()}>
+        Back
+      </button>
+      <span>{tasks.currentPage}</span>
+      <button
+        disabled={nextOffset > totalTasks}
+        onClick={() => handleNextPage()}>
+        Forward
+      </button>
+    </div>
+  );
+}
+
+```
+
+In the next step, we will replace our current pagination method with the cursor pagination method.
```

Comment:
@timDeHof Also add a message here for them to run the code to see what this looks like.

Reply:
<img width="759" alt="Screenshot 2023-10-11 at 4 28 06 PM" src="https://github.com/appwrite/website/assets/29069505/cbcf78ea-3acc-472f-aba6-f4cee5c60913">
Also keep in mind the natural sort order, maybe add a sort query here.